code/go/0chain.net/smartcontract/minersc/miner.go

```diff
@@ -84,6 +87,27 @@ func (msc *MinerSmartContract) AddMiner(t *transaction.Transaction,
 		return "", common.NewError("add_miner",
 			"PublicKey or the ID is empty. Cannot proceed")
 	}
+	
+	// Check delegate wallet is not the same as operational wallet (PUK)
+	if ! config.Development() {
+		publicKeyBytes, err := hex.DecodeString(newMiner.PublicKey)
+		if err != nil {
+			logging.Logger.Error("Couldn't decode public key to compare to delegate wallet")
+			return "", common.NewError("add_miner",
+				"Couldn't decode publick key to compare to delegate wallet")
+		}
+		operationalClientID := encryption.Hash(publicKeyBytes)
+	
+		logging.Logger.Info("comparing delegate wallet",
+			zap.String("delegate_wallet", newMiner.Settings.DelegateWallet), zap.String("operational_wallet", operationalClientID),
+		)
+	
+		if operationalClientID == newMiner.Settings.DelegateWallet {
+			logging.Logger.Error("Can't use the same wallet as both operational and delegate")
+			return "", common.NewError("add_miner",
+				"Can't use the same wallet as both operational and delegate")
```

Comment:
Avoid capital error message, and use consistent message, so change "Can't" to "couldn't" or vice verse. Also I saw repeat code below that basically do the same thing, which we can improve by moving this code to a function, and define in parent package `code/go/0chain.net/smartcontract`.

Reply:
Done